tests/drivers/counter/counter_basic_api/src/test_counter.c

```diff
@@ -295,7 +308,11 @@ void test_all_channels_instance(const char *str)
 
 void test_all_channels(void)
 {
+#ifdef CONFIG_COUNTER_MCUX_RTC
+	ztest_test_skip();
```

Comment:
why you've disabled this test? If device supports only one channel that should work as well.